File: src/api/FleetManager/FleetManager.Application/Vehicles/AddVehicle/AddVehicleRequest.cs
namespace FleetManager.Application.Vehicles.AddVehicle
{
    public record AddVehicleRequest(
        string Vin,
        string LicensePlate,
        string Model,
        DateTime LastInspectionDate,
        DateTime NextInspectionDate,
        string LocationName,
        double Latitude,
        double Longitude)
    {
    }
}

File: src/client/web/FleetManager/src/app/features/vehicles/add-vehicle/models/add-vehicle-request.ts
export class AddVehicleRequest {
  public vin: string;
  public licensePlate: string;
  public model: string;
  public lastInspectionDate: Date;
  public nextInspectionDate: Date;
  public locationName: string;
  public latitude: number;
  public longitude: number;

  constructor(vin: string,
              licensePlate: string,
              model: string,
              lastInspectionDate: Date,
              nextInspectionDate: Date,
              locationName: string,
              latitude: number,
              longitude: number) {
    this.vin = vin;
    this.licensePlate = licensePlate;
    this.model = model;
    this.lastInspectionDate = lastInspectionDate;
    this.nextInspectionDate = nextInspectionDate;
    this.locationName = locationName;
    this.latitude = latitude;
    this.longitude = longitude;
  }
}
